src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -109,6 +108,12 @@ Copyright (c) .NET Foundation. All rights reserved.
                       Version=""
                       IncludeAssets="Runtime;Build;Native;contentFiles;Analyzers"
                       PrivateAssets="All" />
+
+      <!--
+        Add GlobalPackageReference items to the PackageVersion item group with the version.
+      -->
+    <PackageVersion Include="@(GlobalPackageReference)"
```

Comment:
We can probably use a test for this.

Reply:
I will need to pull `MSBuildLocator` into `NuGet.Build.Tasks.Test` so I can do an MSBuild evaluation without calling targets to verify that the transformation is taking place.

The existing unit tests verify that it works end-to-end but doesn't test in the legacy CSPROJ scenario (ie CollectPackageReferences hasn't run).  What do you think?

Reply:
hmmm...I'm wondering if a VS test would be more appropriate? An Apex test for example?

Reply:
What if I make a functional test that just logs the PackageReferences after importing NuGet.props/NuGet.targets?  It would just need to verify that the items and their metadata is set and I could accomplish that with a relatively hacky test.

Reply:
Use evaluated items instead of calling a target?
That definitely tests the problem, but probably not the scenario. 

I don't expect this to often be the problem so though, so I'm fine with that.

Reply:
I have added a unit test that just verifies that the `<GlobalPackageReference />` items are processed without having to call a target.